File: src/Cpp/examples/example01_text_clean.cpp
#include <iostream>
#include <string>

#include "pipeline_services/core/pipeline.hpp"
#include "pipeline_services/examples/text_steps.hpp"

std::string truncate_at_280(std::string text_value, pipeline_services::core::StepControl<std::string>& control) {
  if (text_value.size() <= 280) {
    return text_value;
  }
  control.short_circuit();
  return text_value.substr(0, 280);
}

int main() {
  pipeline_services::core::Pipeline<std::string> pipeline("example01_text_clean", true);
  pipeline.add_action(pipeline_services::examples::strip);
  pipeline.add_action(pipeline_services::examples::normalize_whitespace);
  pipeline.add_action_named("truncate", truncate_at_280);

  const auto result = pipeline.execute("  Hello   World  ");
  std::cout << "output=" << result.context << std::endl;
  std::cout << "shortCircuited=" << (result.short_circuited ? "true" : "false") << std::endl;
  std::cout << "errors=" << result.errors.size() << std::endl;
  return 0;
}

File: src/Rust/examples/example01_text_clean.rs
use pipeline_services::examples::text_steps::{normalize_whitespace, strip};
use pipeline_services::{short_circuit, Pipeline};

fn truncate_at_280(text_value: String) -> String {
  if text_value.chars().count() <= 280 {
    return text_value;
  }
  short_circuit();
  text_value.chars().take(280).collect::<String>()
}

fn main() {
  let mut pipeline = Pipeline::new("example01_text_clean", true);
  pipeline.add_action(strip);
  pipeline.add_action(normalize_whitespace);
  pipeline.add_action_named("truncate", truncate_at_280);

  let result = pipeline.run_detailed("  Hello   World  ".to_string());
  println!("output={}", result.context);
  println!("shortCircuited={}", result.short_circuited);
  println!("errors={}", result.errors.len());
}
